server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v67/MigratePreviousAnalysisLeakPeriodTest.java

```diff
@@ -0,0 +1,188 @@
+package org.sonar.server.platform.db.migration.version.v67;/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import com.google.common.collect.ImmutableList;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.db.CoreDbTester;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MigratePreviousAnalysisLeakPeriodTest {
+
+  private final static String SELECT_PROPERTIES = "SELECT prop_key, is_empty, text_value, clob_value FROM properties";
+
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(MigratePreviousAnalysisLeakPeriodTest.class, "properties.sql");
+
+  private MigratePreviousAnalysisLeakPeriod underTest = new MigratePreviousAnalysisLeakPeriod(db.database());
+
+  @Test
+  public void migration_must_update_the_database() throws SQLException {
+    List<Property> properties = generateAndInsertProperties();
```

Comment:
Not blocker, but I really don't like this pattern (one method at the begin that create users, and one method at the end that do the check) => I prefer to explicitly have the list of properties, as I've no idea which properties you've created.
See for instance [FixEmptyIdentityProviderInUsersTest](https://github.com/SonarSource/sonarqube/blob/master/server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v66/FixEmptyIdentityProviderInUsersTest.java) => you know quickly which users are created, and which data do you expect at the end of the test without having to dig into other method